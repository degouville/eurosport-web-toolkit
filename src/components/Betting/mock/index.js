export const doubleChoiceData = {
  link: 'mainLink',
  sponsor: {
    name: 'Betline',
    backgroundColor: '#b4292c',
    textColor: '#ffffff',
  },
  sentences: ['Pariez !', 'PARIEZ AVEC Betline'],
  choices: [{ number: 1, odds: '2.80', label: 'Nadal' }, { number: 2, odds: '1.70', label: 'Federer' }],
};

export const tripleChoiceData = {
  ...doubleChoiceData,
  choices: [
    { number: 1, odds: '2.80', label: 'Nadal' },
    { number: 2, odds: '4.90' },
    { number: 3, odds: '1.70', label: 'Federer' },
  ],
};

export const alternativeLogo = {
  ...doubleChoiceData,
  sponsor: {
    name: 'AlterBet',
    backgroundColor: '#6decb9',
    textColor: '#3c3c3c',
  },
  sentences: ['Alternate with AlterBet !', 'AlterBet for the BETter'],
};
