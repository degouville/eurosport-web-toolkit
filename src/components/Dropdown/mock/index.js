export const dropdownOptions = [
  { text: 'Men singles', id: 1231 },
  { text: 'Women singles', id: 1232 },
  { text: 'Men doubles', id: 1233 },
  { text: 'Women doubles', id: 1234 },
  { text: 'Men triples', id: 1235 },
  { text: 'Women triples', id: 1236 },
];

export const initialOptionID = dropdownOptions[3].id;

export default {
  ...dropdownOptions,
  ...initialOptionID,
};
