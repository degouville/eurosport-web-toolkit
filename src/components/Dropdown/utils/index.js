// eslint-disable-next-line import/prefer-default-export
export const removeUnit = width => parseInt(width, 10);
