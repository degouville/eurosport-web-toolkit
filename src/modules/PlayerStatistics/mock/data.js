export const names = {
  playerOne: 'A. Mendez-Maceiras',
  playerTwo: 'R. Federer',
};

export const comparisonPoints = [
  {
    id: 223341,
    title: 'ages',
    currentValues: {
      one: 3,
      two: 8,
    },
    maximumValue: 10,
  },
  {
    id: 223342,
    title: 'double faults',
    currentValues: {
      one: 2,
      two: 1,
    },
    maximumValue: 5,
  },
  {
    id: 223343,
    title: 'point won second ball',
    currentValues: {
      one: 84,
      two: 58,
    },
    maximumValue: 100,
    isPercent: true,
  },
  {
    id: 223344,
    title: 'first serve',
    currentValues: {
      one: 61,
      two: 55,
    },
    maximumValue: 100,
    isPercent: true,
  },
  {
    id: 223345,
    title: 'ages',
    currentValues: {
      one: 70,
      two: 41,
    },
    maximumValue: 100,
    isPercent: true,
  },
  {
    id: 223346,
    title: 'point won second ball',
    currentValues: {
      one: 3,
      two: 8,
    },
    maximumValue: 10,
  },
  {
    id: 223347,
    title: 'double faults',
    currentValues: {
      one: 3,
      two: 1,
    },
    maximumValue: 5,
  },
];
