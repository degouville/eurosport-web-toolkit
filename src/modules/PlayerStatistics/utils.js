export const setGradient = ({ firstValue, secondValue, maximumValue, colorOne, colorTwo }) => {
  // NOTE: share values on a range of 100%
  const b = Math.floor(((firstValue / maximumValue) * 100) / 2);
  const c = Math.floor(((secondValue / maximumValue) * 100) / 2);
  const a = 100 / 2 - b;

  const offset = a;
  const playerOne = offset + b;
  const playerTwo = playerOne + c;

  return `
    linear-gradient(
      to right,
      transparent 0%,
      transparent ${offset}%,
      ${colorOne} ${offset}%,
      ${colorOne} ${playerOne}%,
      ${colorTwo} ${playerOne}%,
      ${colorTwo} ${playerTwo}%,
      transparent ${playerTwo}%,
      transparent
    )
  `;
};

export const addPercentMark = (isPercent = false, number) => `${number}${isPercent ? '%' : ''}`;
