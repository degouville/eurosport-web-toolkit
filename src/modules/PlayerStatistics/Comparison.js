import React from 'react';
import styled, { css } from 'react-emotion';
import PropTypes from 'prop-types';
import { H6 } from 'src/typography';
import { theme } from 'src';
import * as breakpoints from 'src/breakpoints';
import { setGradient, addPercentMark } from './utils';

const Comparison = ({ title, currentValues, maximumValue, isPercent, id }) => {
  const { one, two } = currentValues;
  const { colorOne, colorTwo } = theme.statistics.comparison;
  const config = { firstValue: one, secondValue: two, maximumValue, colorOne, colorTwo };
  return (
    <StyledComparison gradient={setGradient(config)} key={id}>
      <StyledValue>{addPercentMark(isPercent, one)}</StyledValue>
      <StyledLabel>{title}</StyledLabel>
      <StyledValue>{addPercentMark(isPercent, two)}</StyledValue>
    </StyledComparison>
  );
};

Comparison.propTypes = {
  title: PropTypes.string.isRequired,
  currentValues: PropTypes.shape(PropTypes.object).isRequired,
  maximumValue: PropTypes.number.isRequired,
  isPercent: PropTypes.bool,
  id: PropTypes.shape(PropTypes.any).isRequired,
};

Comparison.defaultProps = {
  isPercent: false,
};

const StyledComparison = styled.li`
  display: flex;
  justify-content: space-around;
  background: ${props => props.theme.statistics.comparison.background};
  background-image: ${({ gradient }) => gradient};
  margin: 0 0 20px 0;
  &:last-of-type {
    margin: 0;
  }
`;

const ValueCommonStyle = css`
  flex: 1;
  text-align: center;
  text-transform: uppercase;
  min-height: 44px;
  padding: 0 10px;
  display: flex;
  align-items: center;
  justify-content: center;
`;

const StyledValue = styled.p`
  ${ValueCommonStyle}
  font-size: 12px;
  ${breakpoints.medium(css`
    font-size: 14px;
  `)}
`;

const StyledLabel = styled(H6)`
  ${ValueCommonStyle}
  flex: 4;
  font-size: 10px;
  ${breakpoints.small(css`
    flex: 2;
  `)}
  ${breakpoints.medium(css`
    font-size: 12px;
    flex: 1;
  `)}
`;

export default Comparison;
