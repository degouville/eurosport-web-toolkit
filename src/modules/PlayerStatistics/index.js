import React from 'react';
import styled, { css } from 'react-emotion';
import PropTypes from 'prop-types';
import { H3 } from 'src/typography';
import { Grid } from 'src';
import * as breakpoints from 'src/breakpoints';
import ibmLogo from '../../assets/ibm-logo.svg';

import Comparison from './Comparison';

const { Container } = Grid;

const PlayerStatistics = ({ names, comparisonPoints, isSponsored }) => {
  const { playerOne, playerTwo } = names;

  return (
    <StyledContainer>
      <StyledHeader>
        <StyledTitle>{playerOne}</StyledTitle>
        <StyledTitle>{playerTwo}</StyledTitle>
      </StyledHeader>

      <StyledStats>{!!comparisonPoints.length && comparisonPoints.map(Comparison)}</StyledStats>
      {isSponsored && <StyledSponsored src={ibmLogo} />}
    </StyledContainer>
  );
};

PlayerStatistics.defaultProps = {
  isSponsored: false,
};

PlayerStatistics.propTypes = {
  names: PropTypes.shape({
    playerOne: PropTypes.string.isRequired,
    playerTwo: PropTypes.string.isRequired,
  }).isRequired,
  comparisonPoints: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      title: PropTypes.string.isRequired,
      currentValues: PropTypes.shape({
        one: PropTypes.number.isRequired,
        two: PropTypes.number.isRequired,
      }).isRequired,
      maximumValue: PropTypes.number.isRequired,
      isPercent: PropTypes.bool,
    })
  ).isRequired,
  isSponsored: PropTypes.bool,
};

const StyledContainer = styled(Container)`
  margin: 16px;
  color: ${({ theme }) => theme.typo.primary.color};
  background: ${({ theme }) => theme.statistics.background.one};
  background-image: linear-gradient(
    to right,
    ${({ theme }) => theme.statistics.background.one} 0%,
    ${({ theme }) => theme.statistics.background.one} 50%,
    ${({ theme }) => theme.statistics.background.two} 50%,
    ${({ theme }) => theme.statistics.background.two} 100%
  );
  position: relative;
  overflow: hidden;

  &:before {
    content: '';
    position: absolute;
    display: block;
    width: 110%;
    height: 30px;
    ${breakpoints.medium(css`
      height: 60px;
    `)}
    background-image: linear-gradient(
      -183deg,
      ${({ theme }) => theme.statistics.fringe} 0%,
      ${({ theme }) => theme.statistics.fringe} 50%,
      transparent 50%,
      transparent 100%
    );
  }
`;

const StyledHeader = styled.header`
  display: flex;
  justify-content: space-around;
  align-items: center;
`;

const StyledTitle = styled(H3)`
  text-align: center;
  padding: 30px 10px;
  flex: 0 0 calc(50% - 20px);
  word-break: break-word;
  font-size: 18px;
  ${breakpoints.medium(css`
    padding: 50px;
    flex: 0 0 calc(50% - 100px);
    font-size: 24px;
  `)}
`;

const StyledStats = styled.ul`
  padding: 0 15px 15px 15px;
  ${breakpoints.medium(css`
    padding: 0 45px 20px 45px;
  `)}
`;

const StyledSponsored = styled.img`
  height: 26px;
  width: 60px;
  display: block;
  float: right;
  margin: 0 15px 15px 0;
  ${breakpoints.medium(css`
    height: 35px;
    width: 84px;
    margin: 0 45px 20px 0;
  `)}
`;

export default PlayerStatistics;
