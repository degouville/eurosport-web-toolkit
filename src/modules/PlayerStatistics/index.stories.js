import React from 'react';
import { storiesOf } from '@storybook/react';
import { boolean, object } from '@storybook/addon-knobs';
import PlayerStatistics from '.';
import { names, comparisonPoints } from './mock/data';

const playerStatisticsStory = storiesOf('Modules|PlayerStatistics', module);
const render = () => (
  <PlayerStatistics
    names={object('Names', names)}
    comparisonPoints={object('Comparison Points', comparisonPoints)}
    isSponsored={boolean('isSponsored', true)}
  />
);
playerStatisticsStory.add('default', render);
