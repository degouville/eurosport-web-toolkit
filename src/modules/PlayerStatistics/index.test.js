import React from 'react';
import { shallow } from 'enzyme';
import { ThemeProvider } from 'emotion-theming';
import theme from 'src/theme';
import PlayerStatistics from '.';
import { names, comparisonPoints } from './mock/data';

import { setStatsBar, addPercentMark } from './utils';

describe('PlayerStatistics', () => {
  describe('Component', () => {
    it('renders as expected', () => {
      // Given
      const wrapper = shallow(
        <ThemeProvider theme={theme}>
          <PlayerStatistics names={names} comparisonPoints={comparisonPoints} />
        </ThemeProvider>
      );

      const playerStatistics = wrapper.find(PlayerStatistics);

      // Expect
      expect(playerStatistics).toMatchSnapshot();
    });
  });

  describe('Utils', () => {
    it('generate a linear-gradient', () => {
      // Given
      const options = {
        firstValue: 6,
        secondValue: 8,
        maximumValue: 16,
        colorOne: 'blue',
        colorTwo: 'red',
      };

      // When
      const statsBar = setStatsBar({ ...options });

      // Expect
      const wantedBar = `&:before {
        content: '';
        display: block;
        position: absolute;
        right: 50%;
        top: 0;
        bottom: 0;
        width: 18%;
        background: blue;
        transition: width 0.3s ease-out;
      }
      &:after {
        content: '';
        display: block;
        position: absolute;
        left: 50%;
        top: 0;
        bottom: 0;
        width: 25%;
        background: red;
        transition: width 0.3s ease-out;
      }
  `;

      expect(statsBar).toBe(wantedBar);
    });

    it('add percentage mark', () => {
      // Given
      const value = 3;
      const isPercent = true;

      // When
      const normalizedValue = addPercentMark(isPercent, value);

      // Expect
      const wantedValue = `3%`;
      expect(normalizedValue).toBe(wantedValue);
    });
  });
});
