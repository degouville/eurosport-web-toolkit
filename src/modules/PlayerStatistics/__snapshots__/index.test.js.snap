// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerStatistics Component renders as expected 1`] = `
<PlayerStatistics
  comparisonPoints={
    Array [
      Object {
        "currentValues": Object {
          "one": 3,
          "two": 8,
        },
        "id": "223341",
        "maximumValue": 11,
        "title": "ages",
      },
      Object {
        "currentValues": Object {
          "one": 2,
          "two": 1,
        },
        "id": "223342",
        "maximumValue": 3,
        "title": "double faults",
      },
      Object {
        "currentValues": Object {
          "one": 84,
          "two": 58,
        },
        "id": "223343",
        "isPercent": true,
        "maximumValue": 100,
        "title": "point won second ball",
      },
      Object {
        "currentValues": Object {
          "one": 61,
          "two": 55,
        },
        "id": "223344",
        "isPercent": true,
        "maximumValue": 100,
        "title": "first serve",
      },
      Object {
        "currentValues": Object {
          "one": 70,
          "two": 41,
        },
        "id": "223345",
        "isPercent": true,
        "maximumValue": 100,
        "title": "ages",
      },
      Object {
        "currentValues": Object {
          "one": 3,
          "two": 8,
        },
        "id": "223346",
        "maximumValue": 11,
        "title": "point won second ball",
      },
      Object {
        "currentValues": Object {
          "one": 3,
          "two": 1,
        },
        "id": "223347",
        "maximumValue": 4,
        "title": "double faults",
      },
    ]
  }
  isSponsored={false}
  names={
    Object {
      "playerOne": "A. Mendez-Maceiras",
      "playerTwo": "R. Federer",
    }
  }
/>
`;
